addons/sale/tests/test_sale_to_invoice.py

```diff
@@ -921,3 +921,41 @@ def test_partial_invoicing_interaction_with_invoicing_switch_threshold(self):
         self.assertEqual(line.qty_invoiced, 10)
         line.qty_delivered = 15
         self.assertEqual(line.qty_invoiced, 10)
+
+    def test_amount_to_invoice_multiple_so(self):
+        """ Testing creating two SOs with the same customer and invoicing them together. We have to ensure
+            that the amount to invoice is correct for each SO.
+        """
+        sale_order_1 = self.env['sale.order'].create({
+            'partner_id': self.partner_a.id,
+            'order_line': [
+                Command.create({
+                    'product_id': self.company_data['product_delivery_no'].id,
+                    'product_uom_qty': 10,
+                }),
+            ],
+        })
+        sale_order_2 = self.env['sale.order'].create({
+            'partner_id': self.partner_a.id,
+            'order_line': [
+                Command.create({
+                    'product_id': self.company_data['product_delivery_no'].id,
+                    'product_uom_qty': 20,
+                }),
+            ],
+        })
+
+        sale_order_1.action_confirm()
+        sale_order_2.action_confirm()
+        sale_order_1.order_line.qty_delivered = 10
+        sale_order_2.order_line.qty_delivered = 20
+
+        self.env['sale.advance.payment.inv'].create({
+            'advance_payment_method': 'delivered',
+            'sale_order_ids': [(6, 0, [sale_order_1.id, sale_order_2.id])],
```

Comment:
```suggestion
            'sale_order_ids': [Command.set((sale_order_1 + sale_order_2).ids)],
```
You might as well use `Command` everywhere :slightly_smiling_face: